packages/block-library/src/list/save.js

```diff
@@ -7,7 +7,16 @@ export default function save( { attributes } ) {
 	const { ordered, type, reversed, start } = attributes;
 	const TagName = ordered ? 'ol' : 'ul';
 	return (
-		<TagName { ...useBlockProps.save( { type, reversed, start } ) }>
+		<TagName
+			{ ...useBlockProps.save( {
+				reversed,
+				start,
+				style: {
+					listStyleType:
+						ordered && type !== 'decimal' ? type : undefined,
+				},
+			} ) }
+		>
```

Comment:
Why are we changing the save function if we're only needing to fix something in the admin?

Reply:
I think this PR settled on the solution of using the inline style (`list-style-type`) because the `type` attribute of the list would be overridden on the editor side. Shouldn't the save function side also be updated to match the editor side?

Reply:
Yes, sorry, I misunderstood and thought this was the previous fix.

Reply:
How were these styles injected into the frontend before this? Why did we decide to use inline styles now?

Reply:
Because the type attribute is deprecated.

Reply:
Ok Thanks for the details. 👍